### PR TITLE
feat(behavior_path_racing_overtake_module): add module for overtaking in a racing environment

### PR DESCRIPTION
**Pull request by yhisaki**
## Description

A module designed for the purpose of overtaking other cars within a racing circuit. Currently, only simple algorithms have been implemented.

## Related links

see [this video](https://youtu.be/wM9VYNTOtaE)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/config/behavior_path_planner.param.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /**:
   ros__parameters:
-    verbose: false
+    verbose: true
```
Comment (takayuki5168) [inline planning/behavior_path_planner/config/behavior_path_planner.param.yaml:3]: Please revert this.

File: planning/behavior_path_racing_overtake_module/include/behavior_path_racing_overtake_module/state.hpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+class BackToCenter : public RacingOverTakeState
+{
+private:
+  const double after_overtake_shift_length_;
+  PathWithLaneId back_to_center_path_;
+  Pose back_to_center_end_pose_;
+  bool is_first_time_to_plan_ = true;
+
+public:
+  BackToCenter(Context * context, double after_overtake_shift_length)
```
Comment (takayuki5168) [inline planning/behavior_path_racing_overtake_module/include/behavior_path_racing_overtake_module/state.hpp:114]:
```suggestion
  BackToCenter(Context * context, const double after_overtake_shift_length)
```

File: planning/behavior_path_planner/config/behavior_path_planner.param.yaml (hunk 1 above)
Comment (takayuki5168) [inline planning/behavior_path_planner/config/behavior_path_planner.param.yaml:3]:
```suggestion
    verbose: false
```

File: planning/behavior_path_racing_overtake_module/include/behavior_path_racing_overtake_module/util.hpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ * @param reference_path reference path
+ * @param ego_pose ego pose
+ * @param shift_length shift length
+ * @param shift_start_length shift start length
+ * @param shift_end_length shift end length
+ * @return std::pair<PathWithLaneId, Pose> return back to center path and back to center end pose
+ */
+std::pair<PathWithLaneId, Pose> calcBackToCenterPath(
+  const PathWithLaneId & reference_path, const Pose & ego_pose, double shift_length,
+  double shift_start_length, double shift_end_length);
```
[the remark this answers is not in the document]
Comment (takayuki5168) [reply planning/behavior_path_racing_overtake_module/include/behavior_path_racing_overtake_module/util.hpp:83]: Regarding adding the const, please check the remained part of the PR's changes.

File: planning/behavior_path_racing_overtake_module/src/scene.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 TIER IV, Inc.
```
[the remark this answers is not in the document]
Comment (yhisaki) [reply planning/behavior_path_racing_overtake_module/src/scene.cpp]: Thank you for pointing that out. I fixed it.
74bf1d5326a27fe2b722389b832a4429fb1b14d3

File: planning/behavior_path_racing_overtake_module/package.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>behavior_path_racing_overtake_module</name>
+  <version>0.1.0</version>
+  <description>The behavior_path_racing_overtake_module package</description>
+
+  <maintainer email="yhisaki31@gmail.com">Yukinari Hisaki</maintainer>
```
[the remark this answers is not in the document]
Comment (yhisaki) [reply planning/behavior_path_racing_overtake_module/package.xml:10]: Thank you for pointing that out. I fixed it.
58a3f9b343384202cbe2ae1589a806271f0e06ce

[the remark this answers is not in the document]
Comment (yhisaki) [reply planning/behavior_path_racing_overtake_module/package.xml:10]: Thank you for pointing that out. I fixed it.
87c72bb9dbd29ce517bf5acbbd0e80b4be260906

File: planning/behavior_path_racing_overtake_module/include/behavior_path_racing_overtake_module/util.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (yhisaki) [reply planning/behavior_path_racing_overtake_module/include/behavior_path_racing_overtake_module/util.hpp]: Thank you for pointing that out. I fixed it.
87c72bb9dbd29ce517bf5acbbd0e80b4be260906

**Pull request closed without merging**